### PR TITLE
Ft/tests

### PR DESCRIPTION
**Pull request by iamcristos**
# Description
-covers tests for all endpoints
Describe change or new feature here.

## Checklist

Remove any items which are not applicable.

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

File: __test__/users.test.js [hunk 1]
```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
...
+    };
+    return request(server)
+      .post('/api/user/login')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(404);
+        expect(res.body.message).toBe('User not found');
+      });
+  });
+  it('should return 401 if credientials are not valid', () => {
```
Comment (ruihildt) [inline __test__/users.test.js]: Extra `i` to remove in `credentials`.

File: __test__/users.test.js [hunk 2]
```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
...
+    };
+    return request(server)
+      .post('/api/user/login')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(400);
+        expect(res.body.message).toBe('Input a valid email');
+      });
+  });
+  it('should return 404 if user doesn exists', () => {
```
Comment (ruihildt) [inline __test__/users.test.js]: `doesn't`

File: __test__/users.test.js [hunk 3]
```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
...
+      .post('/api/user/signup')
+      .send(user)
+      .then(res => {
+        expect(res.status).toBe(400);
+        expect(res.body.message).toBe(
+          'User already registered with username or email provided'
+        );
+      });
+  });
+  it('should return a 201 when user is created succesful', async () => {
```
Comment (ruihildt) [inline __test__/users.test.js]: `successfully` instead of `successful`

File: __test__/users.test.js [hunk 4]
```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
...
+      .then(res => {
+        expect(res.status).toBe(302);
+      });
+  });
+});
+
+describe('PATCH /api/user/:username/password', () => {
+  it('should return 401 if no token is provided', () => {
+    return request(server)
+      .patch('/api/user/:vincent/password')
```
Comment (ruihildt) [inline __test__/users.test.js]: Do we really need the `:` before the username here?

File: __test__/users.test.js [hunk 5]
```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
...
+      .send({})
+      .then(res => {
+        expect(res.status).toBe(401);
+        expect(res.body.message).toBe('Token is required');
+      });
+  });
+  it('should return 401 if token is invalid', () => {
+    return request(server)
+      .patch('/api/user/:vincent/password')
```
Comment (ruihildt) [inline __test__/users.test.js]: Do we really need the `:` before the username here?

File: __test__/users.test.js [hunk 6]
```diff
@@ -272,3 +275,220 @@ describe('POST user update social media info', () => {
...
+      });
+  });
+  it("should return 404 if user doesn't exists", async () => {
+    const user = {
+      id: 1,
+      username: 'john'
+    };
+    const jwtToken = await jwt.generateToken(user);
+    return request(server)
+      .patch('/api/user/:vincent/password')
```
Comment (ruihildt) [inline __test__/users.test.js]: Do we really need the `:` before the username here?

File: __test__/users.test.js (hunk 4 above)
Comment (iamcristos) [reply __test__/users.test.js]: THIS testing for invalid username

File: __test__/country.test.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+const request = require('supertest');
+const server = require('../server');
+
+describe('GET /api/countries/counties', () => {
```
Comment (ruihildt) [inline __test__/country.test.js]: Typo, missing `r`: GET /api/countries/count**r**ies'

**Pull request merged**